File: app/src/main/java/co/tpcreative/supersafe/ui/theme/ThemeSettingsAdapter.java
package co.tpcreative.supersafe.ui.theme;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import com.bumptech.glide.Glide;
import com.bumptech.glide.Priority;
import com.bumptech.glide.request.RequestOptions;
import butterknife.BindView;
import butterknife.OnClick;
import co.tpcreative.supersafe.R;
import co.tpcreative.supersafe.common.adapter.BaseAdapter;
import co.tpcreative.supersafe.common.adapter.BaseHolder;
import co.tpcreative.supersafe.common.util.Utils;
import co.tpcreative.supersafe.model.Theme;
import de.hdodenhof.circleimageview.CircleImageView;

public class ThemeSettingsAdapter extends BaseAdapter<Theme, BaseHolder> {
    private Context context;
    private ItemSelectedListener itemSelectedListener;
    private String TAG = ThemeSettingsAdapter.class.getSimpleName();
    RequestOptions options = new RequestOptions()
            .centerCrop()
            .override(60,60)
            .priority(Priority.HIGH);

    public ThemeSettingsAdapter(LayoutInflater inflater, Context context, ItemSelectedListener itemSelectedListener) {
        super(inflater);
        this.context = context;
        this.itemSelectedListener = itemSelectedListener;
    }

    @Override
    public int getItemCount() {
        return dataSource.size();
    }

    @Override
    public BaseHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new ItemHolder(inflater.inflate(R.layout.theme_item, parent, false));
    }

    public interface ItemSelectedListener {
        void onClickItem(int position);
    }

    public class ItemHolder extends BaseHolder<Theme> {

        @BindView(R.id.imgTheme)
        CircleImageView imgTheme;
        @BindView(R.id.imgChecked)
        ImageView imgChecked;
        int mPosition;
        Theme theme;

        public ItemHolder(View itemView) {
            super(itemView);
        }

        @Override
        public void bind(Theme data, int position) {
            super.bind(data, position);
            mPosition = position;
            //imgTheme.setBackgroundColor(data.getPrimaryColor());
            Glide.with(context)
                    .load(context.getResources().getDrawable(data.getPrimaryColor()))
                    .apply(options).into(imgTheme);
            theme = data;
            if (data.isCheck){
                imgChecked.setVisibility(View.VISIBLE);
            }
            else {
                imgChecked.setVisibility(View.INVISIBLE);
            }
            Utils.Log(TAG,"Change position "+ position);
        }

        @OnClick(R.id.rlHome)
        public void onClicked(View view) {
            if (itemSelectedListener != null) {
                for (int i = 0;i <dataSource.size();i++){
                    if (dataSource.get(i).isCheck){
                        dataSource.get(i).isCheck = false;
                        notifyItemChanged(i);
                    }
                }
                dataSource.get(mPosition).isCheck = true;
                itemSelectedListener.onClickItem(mPosition);
            }
        }
    }

}

File: app/src/main/java/co/tpcreative/supersafe/ui/theme/ThemeSettingsAdapter.kt
package co.tpcreative.supersafe.ui.theme
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.core.content.ContextCompat
import co.tpcreative.supersafe.R
import co.tpcreative.supersafe.common.adapter.BaseAdapter
import co.tpcreative.supersafe.common.adapter.BaseHolder
import co.tpcreative.supersafe.common.util.Utils
import co.tpcreative.supersafe.model.ThemeApp
import com.bumptech.glide.Glide
import com.bumptech.glide.Priority
import com.bumptech.glide.request.RequestOptions
import kotlinx.android.synthetic.main.theme_item.view.*

class ThemeSettingsAdapter(inflater: LayoutInflater, private val context: Context?, private val itemSelectedListener: ItemSelectedListener?) : BaseAdapter<ThemeApp, BaseHolder<ThemeApp>>(inflater) {
    private val TAG = ThemeSettingsAdapter::class.java.simpleName
    var options: RequestOptions = RequestOptions()
            .centerCrop()
            .override(60, 60)
            .priority(Priority.HIGH)

    override fun getItemCount(): Int {
        return mDataSource.size
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BaseHolder<ThemeApp> {
        return ItemHolder(inflater!!.inflate(R.layout.theme_item, parent, false))
    }

    interface ItemSelectedListener {
        open fun onClickItem(position: Int)
    }

    inner class ItemHolder(itemView: View) : BaseHolder<ThemeApp>(itemView) {
        val imgTheme = itemView.imgTheme
        val imgChecked = itemView.imgChecked
        var mPosition = 0
        var themeApp: ThemeApp? = null
        override fun bind(data: ThemeApp, position: Int) {
            super.bind(data, position)
            mPosition = position
            Glide.with(context!!)
                    .load(ContextCompat.getDrawable(context,data.getPrimaryColor()))
                    .apply(options).into(imgTheme!!)
            themeApp = data
            if (data.isCheck) {
                imgChecked?.visibility = View.VISIBLE
            } else {
                imgChecked?.visibility = View.INVISIBLE
            }
            Utils.Log(TAG, "Change position $position")
            itemView.rlHome.setOnClickListener {
                if (itemSelectedListener != null) {
                    for (i in mDataSource.indices) {
                        if (mDataSource[i].isCheck) {
                            mDataSource[i].isCheck = false
                            notifyItemChanged(i)
                        }
                    }
                    mDataSource[mPosition].isCheck = true
                    itemSelectedListener.onClickItem(mPosition)
                }
            }
        }
    }
}